ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -181,31 +199,69 @@ azure_add_reference(WT_STORAGE_SOURCE *storage_source)
     return 0;
 }
 
+// Flush a file locally to Azure.
 static int
 azure_flush(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session, WT_FILE_SYSTEM *file_system,
   const char *source, const char *object, const char *config)
 {
     WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(file_system);
-    WT_UNUSED(source);
-    WT_UNUSED(object);
     WT_UNUSED(config);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+    WT_FILE_SYSTEM *wt_filesystem = azure_fs->wt_fs;
 
-    return 0;
+    std::string src_path = azure_path(azure_fs->home_dir, source);
+    bool exists_native = false;
+
+    int ret = wt_filesystem->fs_exist(wt_filesystem, session, src_path.c_str(), &exists_native);
+    if (ret != 0) {
+        std::cerr << "azure_flush: Failed to check for the existence of " + std::string(source) +
+            " on the native filesystem."
+                  << std::endl;
+        return ret;
+    }
+
+    if (!exists_native) {
+        std::cerr << "azure_flush: " + std::string(source) + " No such file." << std::endl;
+        return ENOENT;
+    }
+
+    std::cout << "azure_flush: Uploading object: " + std::string(object) +
+        " into bucket using put_object"
+              << std::endl;
+
+    // Upload the object into the bucket.
+    ret = azure_fs->azure_conn->put_object(object, src_path);
+    if (ret != 0)
+        std::cerr << "azure_flush: Put object request to Azure failed." << std::endl;
+    else
+        std::cout << "azure_flush: Uploaded object to Azure." << std::endl;
+    return ret;
 }
 
+// Check that flush has been completed by checking the object exists in Azure.
 static int
 azure_flush_finish(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
   WT_FILE_SYSTEM *file_system, const char *source, const char *object, const char *config)
 {
     WT_UNUSED(storage_source);
     WT_UNUSED(session);
-    WT_UNUSED(file_system);
-    WT_UNUSED(source);
-    WT_UNUSED(object);
     WT_UNUSED(config);
+    WT_UNUSED(source);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
 
+    std::cout << "azure_flush_finish: Checking object: " + std::string(object) + " exists in Azure."
+              << std::endl;
+
+    // Check whether the object exists in the cloud.
+    bool exists_cloud = false;
+    azure_fs->azure_conn->object_exists(object, exists_cloud);
+    if (exists_cloud)
```

Comment:
We should return an error if it doesn't exist the cloud.

Reply:
yup i've forgotten this. Thanks for the reminder